doc: Added nRF54H20 migration guides for NCS 2.7 (#15760)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_gs.rst

Hardware
========

* nRF54H20 DK version PCA10175 v0.7.x or v0.8.0 (ES3), ES4.

>   These are the only versions of the nRF54H20 DK compatible with |NCS| v2.7.0.
>   Check the version number on your DK's sticker to verify its compatibility with |NCS| version v2.7.0.

Reviewer: ```suggestion
  These are the only versions of the nRF54H20 DK compatible with the |NCS| v2.7.0.
  Check the version number on your DK's sticker to verify its compatibility with the |NCS| version v2.7.0.
```

* USB-C cable.

Software
========

On your computer, one of the following operating systems:

.. include:: ../../../../nrf/installation/recommended_versions.rst
    :start-after: os_table_start
    :end-before: os_table_end

See :ref:`supported_OS` for more information about the tier definitions.

|supported OS|

You also need the following:

* `Git`_ or `Git for Windows`_ (on Linux and Mac, or Windows, respectively).
* `curl`_.
* The latest version of the `nRF Command Line Tools`_ package.

>   After downloading and installing the tools, add the nrfjprog executable to the system path, on Linux and MacOS, or to the environment variables, on Windows, to run it from anywhere on the system.

Reviewer: ```suggestion
  After downloading and installing the tools, add the nrfjprog executable to the system path on Linux and MacOS, or to the environment variables on Windows. 
  This allows you to run it from anywhere on the system.
```

* On Windows, SEGGER USB Driver for J-Link from SEGGER `J-Link version 7.94e`_.

   .. note::
      To install the SEGGER USB Driver for J-Link on Windows, you must manually reinstall J-Link v7.94e from the command line using the ``-InstUSBDriver=1`` parameter, updating the installation previously run by the `nRF Command Line Tools`_:

      1. Navigate to the download location of the J-Link executable and run one of the following commands:

         * From the Command Prompt::

            JLink_Windows_V794e_x86_64.exe -InstUSBDriver=1

         * From PowerShell::

            .\JLink_Windows_V794e_x86_64.exe -InstUSBDriver=1

      #. In the :guilabel:`Choose optional components` window, select :guilabel:`update existing installation`.

>       #. Add the J-Link executable to the system path, on Linux and MacOS, or to the environment variables, on Windows, to run it from anywhere on the system.

Reviewer: ```suggestion
      #. Add the J-Link executable to the system path on Linux and MacOS, or to the environment variables on Windows, to run it from anywhere on the system.
```

Hardware
========

* nRF54H20 DK version PCA10175 v0.7.x or v0.8.0 (ES3), ES4.

>   These are the only versions of the nRF54H20 DK compatible with |NCS| 2.7.
>   Check the version number on your DK's sticker to verify its compatibility with |NCS| version 2.7.

Reviewer 2: ```suggestion
  These are the only versions of the nRF54H20 DK compatible with |NCS| v2.7.0.
  Check the version number on your DK's sticker to verify its compatibility with |NCS| v2.7.0.
```
Several places where the version number is wrong on other pages as well.

.. _ug_nrf54h20_install_toolchain:

Installing the |NCS| v2.7.0 and its toolchain
*********************************************

You can install the |NCS| v2.7.0 and its toolchain by using Toolchain Manager.

Toolchain Manager is a tool available from `nRF Connect for Desktop`_, a cross-platform tool that provides different applications that simplify installing the |NCS|.

> Both the tool and the application are available for Windows, Linux, and MacOS.

Reviewer 3: Last time I checked with @ihansse , Toolchain Manager is recommended only for NCS v1.9.0 and earlier, and so we state in the docs about it: https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/installation/install_ncs.rst#legacy-installation-with-toolchain-manager

Having this information here contradicts the info I got from Ingar, so I assume it is incorrect.
Also, you have all the installation instructions on the page above I linked.

Author: This is valid only for lilium, and supposedly just for 2.7

         1. With admin permissions enabled, download and install the `nRF Command Line Tools`_.
         #. Restart the Toolchain Manager application.
         #. Click the dropdown menu for the installed nRF Connect SDK version.

            .. figure:: ../../../../nrf/installation/images/gs-assistant_tm_dropdown.png
               :alt: The Toolchain Manager dropdown menu for the installed nRF Connect SDK version, cropped

               The Toolchain Manager dropdown menu options

         #. Select :guilabel:`Open command prompt`.

         You can then follow the instructions in :ref:`creating_cmd`.

Installing the Terminal application
***********************************

On your computer, install `nRF Connect for Desktop`_.
You must also install a terminal emulator, such as `nRF Connect Serial Terminal`_ (from the nRF Connect for Desktop application) or the nRF Terminal (part of the `nRF Connect for Visual Studio Code`_ extension).

Installing nRF Util and its commands
************************************

Using the nRF54H20 DK with the |NCS| v2.7.0 requires the following:

* nRF Util version 7.11.1 or above

> * nRF Util ``device`` version 2.4.0

Reviewer 3: Link here: https://nordic-dev.zoominsoftware.io/bundle/nrfutil/page/nrfutil-device/guides/programming.html

Reviewer 3: I still think this link wouldn't be a bad idea.

1. Download the nrfutil executable file from the `nRF Util development tool`_ product page.

> #. Add nRF Util to the system path (on Linux and MacOS) or environment variables (on Windows) to run it from anywhere on the system.

Reviewer: ```suggestion
#. Add nRF Util to the system path on Linux and MacOS, or environment variables on Windows, to run it from anywhere on the system.
```
alignment

   On Linux and MacOS, use one of the following options:

   * Add nRF Util's directory to the system path.
   * Move the file to a directory in the system path.

#. On MacOS and Linux, give ``nrfutil`` execute permissions by typing ``chmod +x nrfutil`` in a terminal or using a file browser.
   This is typically a checkbox found under file properties.
#. On MacOS, To run the nrfutil executable you need to allow it in the system settings.
#. Verify the version of the nRF Util installation on your machine by running the following command::

      nrfutil --version

#. If your version is below 7.11.1, run the following command to update nRF Util::

>       nrfutil self-upgrade

Reviewer 3: Please link here instead: https://nordic-dev.zoominsoftware.io/bundle/nrfutil/page/guides/installing.html

What is the point of me moving the nRF Util docs and updating them on weekly basis if we just duplicate info over and over?

Author: The getting started guide is supposed to be a one-stop shop for getting started with lilium. This section was tested during the last UX testing. I would not edit this for this release. We can unify this after 2.7

#. Install the nRF Util ``device`` command to version 2.4.0 as follows::

>       nrfutil install device=2.4.0 --force

Reviewer 3: In addition, link here: https://nordic-dev.zoominsoftware.io/bundle/nrfutil/page/guides/installing_commands.html

I'll make sure installing with ``--force`` is documented there before v2.7.0 is live.

Reviewer 4: If you want to ensure that exactly version 2.4.0 is installed, then `--force` is needed. If `--force` is not specified, then this command will only install nrfutil-device if it is not yet installed, and not do anything if any version of nrfutil-device is already installed.
